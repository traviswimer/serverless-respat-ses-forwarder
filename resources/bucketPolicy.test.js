const bucketPolicy = require('./bucketPolicy');

describe('bucketPolicy()', () => {
	test('generates policy', () => {
		expect(bucketPolicy({prefix: "ThePrefix", pattern_name: "ThePatternName"})).toMatchSnapshot();
	});
});
