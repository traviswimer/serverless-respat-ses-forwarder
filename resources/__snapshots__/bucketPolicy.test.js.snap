// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucketPolicy() generates policy 1`] = `
Object {
  "DependsOn": Array [
    "ThePatternNameEmailBucket",
  ],
  "Properties": Object {
    "Bucket": Object {
      "Fn::Join": Array [
        "",
        Array [
          Object {
            "Ref": "ThePatternNameEmailBucket",
          },
        ],
      ],
    },
    "PolicyDocument": Object {
      "Statement": Array [
        Object {
          "Action": "s3:PutObject",
          "Condition": Object {
            "StringEquals": Object {
              "aws:Referer": Object {
                "Ref": "AWS::AccountId",
              },
            },
          },
          "Effect": "Allow",
          "Principal": Object {
            "Service": "ses.amazonaws.com",
          },
          "Resource": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:aws:s3:::",
                Object {
                  "Ref": "ThePatternNameEmailBucket",
                },
                "/*",
              ],
            ],
          },
          "Sid": "ThePrefix-ThePatternName-allow-ses-to-write",
        },
      ],
      "Version": "2012-10-17",
    },
  },
  "Type": "AWS::S3::BucketPolicy",
}
`;
